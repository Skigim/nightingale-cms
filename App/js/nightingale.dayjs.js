/**
 * Nightingale Application Suite - Shared Day.js Utility Wrapper
 *
 * This file provides a standardized, globally accessible 'dateUtils' object
 * that wraps the Day.js library. This ensures consistent date and time
 * formatting and manipulation across all applications in the suite.
 */

// The Day.js library and its plugins are expected to be loaded globally before this script.
const dateUtils = {
  /**
   * Returns the current timestamp as an ISO 8601 string.
   * @returns {string} The current date and time in ISO format.
   */
  now: () => dayjs().toISOString(),

  /**
   * Formats a date string into a specified format.
   * @param {string} dateString - The date string to format.
   * @param {string} formatStr - The desired output format (e.g., "MM/DD/YYYY").
   * @returns {string} The formatted date string, or "N/A" if invalid.
   */
  format: (dateString, formatStr = "MM/DD/YYYY") => {
    if (!dateString) return "N/A";
    const date = dayjs(dateString);
    return date.isValid() ? date.format(formatStr) : "N/A";
  },

  /**
   * Compares two dates for sorting. Returns -1 if A is before B, 1 if A is after B, 0 if equal.
   * Handles invalid dates gracefully.
   * @param {string} dateA - The first date string.
   * @param {string} dateB - The second date string.
   * @returns {number} The comparison result (-1, 0, or 1).
   */
  compareDates: (dateA, dateB) => {
    const d1 = dayjs(dateA);
    const d2 = dayjs(dateB);
    if (!d1.isValid() || !d2.isValid()) return 0;
    if (d1.isBefore(d2)) return -1;
    if (d1.isAfter(d2)) return 1;
    return 0;
  },

  /**
   * Checks if the first date is before the second date.
   * @param {string} dateToCheck The date to evaluate.
   * @param {string} dateToCompareAgainst The baseline date.
   * @returns {boolean}
   */
  isBefore: (dateToCheck, dateToCompareAgainst) => {
    const d1 = dayjs(dateToCheck);
    const d2 = dayjs(dateToCompareAgainst);
    return d1.isValid() && d2.isValid() ? d1.isBefore(d2) : false;
  },

  /**
   * Calculates a date that was a specified number of months ago from now.
   * @param {number} months The number of months to subtract.
   * @returns {string} The resulting date in ISO format.
   */
  monthsAgo: (months) => dayjs().subtract(months, 'month').toISOString(),
};