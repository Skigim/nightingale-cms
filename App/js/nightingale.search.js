/**
 * Nightingale Application Suite - Shared Search Service
 *
 * This service provides a standardized, reusable wrapper around the Fuse.js
 * library for powerful, client-side fuzzy searching of application data.
 */
class NightingaleSearchService {
  /**
   * Creates an instance of the search service.
   * @param {Array<Object>} documents The array of objects to search (e.g., state.cases).
   * @param {Object} options The Fuse.js configuration options.
   * - keys: An array of strings representing the object keys to search in.
   * - threshold: A number from 0.0 to 1.0 indicating search sensitivity.
   */
  constructor(documents = [], options = {}) {
    const defaultOptions = {
      keys: [],
      includeScore: false,
      threshold: 0.3, // A good starting point for general purpose search
      ignoreLocation: true,
    };

    this.fuse = new Fuse(documents, { ...defaultOptions, ...options });
  }

  /**
   * Performs a search against the documents.
   * @param {string} query The search term.
   * @returns {Array<Object>} An array of the matching original objects.
   */
  search(query) {
    if (!query || !query.trim()) {
      // If the query is empty, return the full, original list.
      return this.fuse.getIndex().docs;
    }
    // Fuse.js returns results in a specific format; we map it to return just the original items.
    return this.fuse.search(query).map(result => result.item);
  }

  /**
   * Updates the entire collection of documents to be searched.
   * Useful when the underlying state data changes.
   * @param {Array<Object>} newDocuments The new array of objects.
   */
  updateCollection(newDocuments) {
    this.fuse.setCollection(newDocuments);
  }
}