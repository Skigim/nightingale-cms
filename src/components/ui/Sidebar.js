/**
 * Sidebar.js - Navigation sidebar UI component
 * 
 * Generic UI component for application navigation with tab switching and settings access.
 * Provides a compact sidebar with icons, tooltips, and special states.
 * 
 * @namespace NightingaleUI
 * @version 1.0.0
 * @author Nightingale CMS Team
 */

/**
 * Sidebar Component
 * Displays a vertical navigation sidebar with tabs and settings button
 * 
 * @param {Object} props - Component props
 * @param {string} props.activeTab - ID of the currently active tab
 * @param {Function} props.onTabChange - Callback when a tab is selected
 * @param {Function} [props.onSettingsClick] - Callback when settings button is clicked
 * @param {string} [props.caseViewMode] - Current view mode for cases (e.g., 'details')
 * @param {Function} [props.onCaseBackToList] - Callback to navigate back to cases list
 * @returns {React.Element} Sidebar component
 */
function Sidebar({ activeTab, onTabChange, onSettingsClick, caseViewMode, onCaseBackToList }) {
  // React safety check
  if (!window.React) {
    console.warn('React not available for Sidebar component');
    return null;
  }

  const e = window.React.createElement;

  // Validate required props
  if (!activeTab || typeof onTabChange !== 'function') {
    console.warn('Sidebar: activeTab and onTabChange props are required');
    return null;
  }

  const tabs = [
    {
      id: 'dashboard',
      label: 'Dashboard',
      icon: 'M3 7v10a2 2 0 002 2h14a2 2 0 002-2V9a2 2 0 00-2-2H5a2 2 0 00-2-2z',
    },
    {
      id: 'cases',
      label: 'Cases',
      icon: 'M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z',
    },
    {
      id: 'people',
      label: 'People',
      icon: 'M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0z',
    },
    {
      id: 'organizations',
      label: 'Organizations',
      icon: 'M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4',
    },
    {
      id: 'eligibility',
      label: 'Eligibility',
      icon: 'M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z',
    },
  ];

  return e(
    'div',
    {
      className: 'w-16 bg-gray-800 border-r border-gray-700 flex flex-col',
    },
    e(
      'div',
      { className: 'p-4 border-b border-gray-700' },
      e('h1', { className: 'text-blue-400 font-bold text-sm text-center' }, 'CMS')
    ),
    e(
      'nav',
      { className: 'flex-1 p-2 space-y-2' },
      tabs.map((tab) => {
        // Special handling for Cases tab when in details view
        if (tab.id === 'cases' && activeTab === 'cases' && caseViewMode === 'details') {
          return e(
            'button',
            {
              key: tab.id,
              onClick: () => onCaseBackToList && onCaseBackToList(),
              className:
                'w-full p-3 rounded-md transition-all duration-200 group relative bg-blue-600 text-white',
              title: 'Back to Cases List',
            },
            e(
              'svg',
              {
                className: 'w-5 h-5 mx-auto',
                fill: 'none',
                viewBox: '0 0 24 24',
                stroke: 'currentColor',
                strokeWidth: 2,
              },
              e('path', {
                strokeLinecap: 'round',
                strokeLinejoin: 'round',
                d: 'M10 19l-7-7m0 0l7-7m-7 7h18',
              })
            ),
            // Tooltip
            e(
              'div',
              {
                className:
                  'absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50',
              },
              'Back to Cases List'
            )
          );
        }

        // Regular tab rendering
        return e(
          'button',
          {
            key: tab.id,
            onClick: () => onTabChange(tab.id),
            className: `w-full p-3 rounded-md transition-all duration-200 group relative ${
              activeTab === tab.id ? 'bg-blue-600 text-white' : 'text-gray-400 hover:bg-gray-700 hover:text-white'
            }`,
            title: tab.label,
          },
          e(
            'svg',
            {
              className: 'w-5 h-5 mx-auto',
              fill: 'none',
              viewBox: '0 0 24 24',
              stroke: 'currentColor',
              strokeWidth: 2,
            },
            e('path', {
              strokeLinecap: 'round',
              strokeLinejoin: 'round',
              d: tab.icon,
            })
          ),
          // Tooltip
          e(
            'div',
            {
              className:
                'absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50',
            },
            tab.label
          )
        );
      })
    ),
    // Settings button
    onSettingsClick && e(
      'div',
      { className: 'p-2 border-t border-gray-700' },
      e(
        'button',
        {
          onClick: onSettingsClick,
          className:
            'w-full p-3 rounded-md text-gray-400 hover:bg-gray-700 hover:text-white transition-all duration-200 group relative',
          title: 'Settings',
        },
        e(
          'svg',
          {
            className: 'w-5 h-5 mx-auto',
            fill: 'none',
            viewBox: '0 0 24 24',
            stroke: 'currentColor',
            strokeWidth: 2,
          },
          e('path', {
            strokeLinecap: 'round',
            strokeLinejoin: 'round',
            d: 'M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.096 2.572-1.065z',
          }),
          e('path', {
            strokeLinecap: 'round',
            strokeLinejoin: 'round',
            d: 'M15 12a3 3 0 11-6 0 3 3 0 016 0z',
          })
        ),
        // Tooltip
        e(
          'div',
          {
            className:
              'absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50',
          },
          'Settings'
        )
      )
    )
  );
}

// PropTypes for validation
if (typeof window !== 'undefined' && window.PropTypes) {
  Sidebar.propTypes = {
    activeTab: window.PropTypes.string.isRequired,
    onTabChange: window.PropTypes.func.isRequired,
    onSettingsClick: window.PropTypes.func,
    caseViewMode: window.PropTypes.string,
    onCaseBackToList: window.PropTypes.func,
  };
}

// Self-registration for both module and script loading
if (typeof window !== 'undefined') {
  // Register globally for backward compatibility
  window.Sidebar = Sidebar;

  // Register with NightingaleUI registry if available
  if (window.NightingaleUI) {
    window.NightingaleUI.registerComponent('Sidebar', Sidebar, 'layout');
  }
}

// Export for ES6 module compatibility
if (typeof module !== 'undefined' && module.exports) {
  module.exports = Sidebar;
}