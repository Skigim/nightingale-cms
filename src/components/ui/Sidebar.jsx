import React from 'react';
import PropTypes from 'prop-types';
import { registerComponent } from '../../services/registry';

/**
 * Sidebar.js - Navigation sidebar UI component
 *
 * Generic UI component for application navigation with tab switching and settings access.
 * Provides a compact sidebar with icons, tooltips, and special states.
 *
 * @version 1.0.0
 * @author Nightingale CMS Team
 */

/**
 * Sidebar Component
 * Displays a vertical navigation sidebar with tabs and settings button
 *
 * @param {Object} props - Component props
 * @param {string} props.activeTab - ID of the currently active tab
 * @param {Function} props.onTabChange - Callback when a tab is selected
 * @param {Function} [props.onSettingsClick] - Callback when settings button is clicked
 * @param {string} [props.caseViewMode] - Current view mode for cases (e.g., 'details')
 * @param {Function} [props.onCaseBackToList] - Callback to navigate back to cases list
 * @returns {React.Element} Sidebar component
 */
function Sidebar({
  activeTab,
  onTabChange,
  onSettingsClick,
  onReportBugClick,
  caseViewMode,
  onCaseBackToList,
}) {
  // Validate required props
  if (!activeTab || typeof onTabChange !== 'function') {
    return null;
  }

  const tabs = [
    {
      id: 'dashboard',
      label: 'Dashboard',
      icon: 'M3 7v10a2 2 0 002 2h14a2 2 0 002-2V9a2 2 0 00-2-2H5a2 2 0 00-2-2z',
    },
    {
      id: 'cases',
      label: 'Cases',
      icon: 'M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z',
    },
    {
      id: 'people',
      label: 'People',
      icon: 'M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0z',
    },
    {
      id: 'organizations',
      label: 'Organizations',
      icon: 'M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4',
    },
    {
      id: 'eligibility',
      label: 'Eligibility',
      icon: 'M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z',
    },
  ];

  const renderTabButton = (tab) => {
    // Special handling for Cases tab when in details view
    if (
      tab.id === 'cases' &&
      activeTab === 'cases' &&
      caseViewMode === 'details'
    ) {
      return (
        <button
          key={tab.id}
          onClick={() => onCaseBackToList && onCaseBackToList()}
          className="w-full p-3 rounded-md transition-all duration-200 group relative bg-blue-600 text-white"
          title="Back to Cases List"
        >
          <svg
            className="w-5 h-5 mx-auto"
            fill="none"
            viewBox="0 0 24 24"
            stroke="currentColor"
            strokeWidth={2}
          >
            <path
              strokeLinecap="round"
              strokeLinejoin="round"
              d="M10 19l-7-7m0 0l7-7m-7 7h18"
            />
          </svg>
          {/* Tooltip */}
          <div className="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50">
            Back to Cases List
          </div>
        </button>
      );
    }

    // Regular tab rendering
    return (
      <button
        key={tab.id}
        onClick={() => onTabChange(tab.id)}
        className={`w-full p-3 rounded-md transition-all duration-200 group relative ${
          activeTab === tab.id
            ? 'bg-blue-600 text-white'
            : 'text-gray-400 hover:bg-gray-700 hover:text-white'
        }`}
        title={tab.label}
      >
        <svg
          className="w-5 h-5 mx-auto"
          fill="none"
          viewBox="0 0 24 24"
          stroke="currentColor"
          strokeWidth={2}
        >
          <path
            strokeLinecap="round"
            strokeLinejoin="round"
            d={tab.icon}
          />
        </svg>
        {/* Tooltip */}
        <div className="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50">
          {tab.label}
        </div>
      </button>
    );
  };

  return (
    <div className="w-16 bg-gray-800 border-r border-gray-700 flex flex-col">
      <div className="p-4 border-b border-gray-700">
        <h1 className="text-blue-400 font-bold text-sm text-center">CMS</h1>
      </div>

      <nav className="flex-1 p-2 space-y-2">{tabs.map(renderTabButton)}</nav>

      {/* Report Bug button (above settings) */}
      {onReportBugClick && (
        <div className="p-2 border-t border-gray-700">
          <button
            onClick={onReportBugClick}
            className="w-full p-3 rounded-md text-yellow-400 hover:bg-gray-700 hover:text-yellow-300 transition-all duration-200 group relative"
            title="Report a Bug"
          >
            <svg
              className="w-5 h-5 mx-auto"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
              strokeWidth={2}
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                d="M12 9v2m0 4h.01M5 13a7 7 0 1114 0A7 7 0 015 13z"
              />
            </svg>
            {/* Tooltip */}
            <div className="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50">
              Report a Bug
            </div>
          </button>
        </div>
      )}

      {/* Settings button */}
      {onSettingsClick && (
        <div className="p-2 border-t border-gray-700">
          <button
            onClick={onSettingsClick}
            className="w-full p-3 rounded-md text-gray-400 hover:bg-gray-700 hover:text-white transition-all duration-200 group relative"
            title="Settings"
          >
            <svg
              className="w-5 h-5 mx-auto"
              fill="none"
              viewBox="0 0 24 24"
              stroke="currentColor"
              strokeWidth={2}
            >
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                d="M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.096 2.572-1.065z"
              />
              <path
                strokeLinecap="round"
                strokeLinejoin="round"
                d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              />
            </svg>
            {/* Tooltip */}
            <div className="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50">
              Settings
            </div>
          </button>
        </div>
      )}
    </div>
  );
}

Sidebar.propTypes = {
  activeTab: PropTypes.string.isRequired,
  onTabChange: PropTypes.func.isRequired,
  onSettingsClick: PropTypes.func,
  onReportBugClick: PropTypes.func,
  caseViewMode: PropTypes.string,
  onCaseBackToList: PropTypes.func,
};

// Register component
registerComponent('ui', 'Sidebar', Sidebar);

export default Sidebar;
