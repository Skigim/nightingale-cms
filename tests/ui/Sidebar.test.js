/**
 * @jest-environment jsdom
 */

import { render, screen, fireEvent } from '@testing-library/react';
import React from 'react';

// Import the component
require('../Sidebar.js');

describe('Sidebar Component', () => {
  beforeEach(() => {
    // Ensure React is available globally
    global.window = window;
    window.React = React;
  });

  test('renders without crashing', () => {
    const Component = window.Sidebar;
    expect(Component).toBeDefined();

    render(
      React.createElement(Component, {
        activeTab: 'dashboard',
        onTabChange: jest.fn(),
      })
    );
  });

  test('renders sidebar with navigation tabs', () => {
    const Component = window.Sidebar;

    render(
      React.createElement(Component, {
        activeTab: 'dashboard',
        onTabChange: jest.fn(),
      })
    );

    // Check for CMS title
    expect(screen.getByText('CMS')).toBeInTheDocument();

    // Check for tab buttons by their title attributes
    expect(screen.getByTitle('Dashboard')).toBeInTheDocument();
    expect(screen.getByTitle('Cases')).toBeInTheDocument();
    expect(screen.getByTitle('People')).toBeInTheDocument();
    expect(screen.getByTitle('Organizations')).toBeInTheDocument();
    expect(screen.getByTitle('Eligibility')).toBeInTheDocument();
  });

  test('highlights active tab correctly', () => {
    const Component = window.Sidebar;

    render(
      React.createElement(Component, {
        activeTab: 'cases',
        onTabChange: jest.fn(),
      })
    );

    const casesTab = screen.getByTitle('Cases');
    expect(casesTab).toHaveClass('bg-blue-600');
  });

  test('calls onTabChange when tab is clicked', () => {
    const mockOnTabChange = jest.fn();
    const Component = window.Sidebar;

    render(
      React.createElement(Component, {
        activeTab: 'dashboard',
        onTabChange: mockOnTabChange,
      })
    );

    const peopleTab = screen.getByTitle('People');
    fireEvent.click(peopleTab);

    expect(mockOnTabChange).toHaveBeenCalledWith('people');
  });

  test('shows tooltips on hover', () => {
    const Component = window.Sidebar;

    render(
      React.createElement(Component, {
        activeTab: 'dashboard',
        onTabChange: jest.fn(),
      })
    );

    const dashboardTab = screen.getByTitle('Dashboard');
    expect(dashboardTab).toHaveAttribute('title', 'Dashboard');
  });

  test('handles collapsed state', () => {
    const Component = window.Sidebar;

    render(
      React.createElement(Component, {
        activeTab: 'dashboard',
        onTabChange: jest.fn(),
      })
    );

    // Current component always renders in collapsed (w-16) state
    const sidebar = document.querySelector('.w-16');
    expect(sidebar).toBeInTheDocument();
  });

  test('handles settings click', () => {
    const mockOnSettingsClick = jest.fn();
    const Component = window.Sidebar;

    render(
      React.createElement(Component, {
        activeTab: 'dashboard',
        onTabChange: jest.fn(),
        onSettingsClick: mockOnSettingsClick,
      })
    );

    // Settings functionality would be implemented when component is enhanced
    // For now, just verify component renders with settings prop
    expect(Component).toBeDefined();
  });

  test('handles missing required props gracefully', () => {
    const Component = window.Sidebar;

    const { container } = render(React.createElement(Component, {}));

    expect(container.firstChild).toBeNull();
  });

  test('matches snapshot', () => {
    const Component = window.Sidebar;

    const { container } = render(
      React.createElement(Component, {
        activeTab: 'cases',
        onTabChange: jest.fn(),
      })
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
