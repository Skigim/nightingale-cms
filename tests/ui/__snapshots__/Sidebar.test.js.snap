// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Component matches snapshot 1`] = `
<div
  class="w-16 bg-gray-800 border-r border-gray-700 flex flex-col"
>
  <div
    class="p-4 border-b border-gray-700"
  >
    <h1
      class="text-blue-400 font-bold text-sm text-center"
    >
      CMS
    </h1>
  </div>
  <nav
    class="flex-1 p-2 space-y-2"
  >
    <button
      class="w-full p-3 rounded-md transition-all duration-200 group relative text-gray-400 hover:bg-gray-700 hover:text-white"
      title="Dashboard"
    >
      <svg
        class="w-5 h-5 mx-auto"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 7v10a2 2 0 002 2h14a2 2 0 002-2V9a2 2 0 00-2-2H5a2 2 0 00-2-2z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50"
      >
        Dashboard
      </div>
    </button>
    <button
      class="w-full p-3 rounded-md transition-all duration-200 group relative bg-blue-600 text-white"
      title="Cases"
    >
      <svg
        class="w-5 h-5 mx-auto"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 12h6m-6 4h6m2 5H7a2 2 0 01-2-2V5a2 2 0 012-2h5.586a1 1 0 01.707.293l5.414 5.414a1 1 0 01.293.707V19a2 2 0 01-2 2z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50"
      >
        Cases
      </div>
    </button>
    <button
      class="w-full p-3 rounded-md transition-all duration-200 group relative text-gray-400 hover:bg-gray-700 hover:text-white"
      title="People"
    >
      <svg
        class="w-5 h-5 mx-auto"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 20h5v-2a3 3 0 00-5.356-1.857M17 20H7m10 0v-2c0-.656-.126-1.283-.356-1.857M7 20H2v-2a3 3 0 015.356-1.857M7 20v-2c0-.656.126-1.283.356-1.857m0 0a5.002 5.002 0 019.288 0M15 7a3 3 0 11-6 0 3 3 0 016 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50"
      >
        People
      </div>
    </button>
    <button
      class="w-full p-3 rounded-md transition-all duration-200 group relative text-gray-400 hover:bg-gray-700 hover:text-white"
      title="Organizations"
    >
      <svg
        class="w-5 h-5 mx-auto"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-5m-9 0H3m2 0h5M9 7h1m-1 4h1m4-4h1m-1 4h1m-5 10v-5a1 1 0 011-1h2a1 1 0 011 1v5m-4 0h4"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50"
      >
        Organizations
      </div>
    </button>
    <button
      class="w-full p-3 rounded-md transition-all duration-200 group relative text-gray-400 hover:bg-gray-700 hover:text-white"
      title="Eligibility"
    >
      <svg
        class="w-5 h-5 mx-auto"
        fill="none"
        stroke="currentColor"
        stroke-width="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      <div
        class="absolute left-full ml-2 px-2 py-1 bg-gray-900 text-white text-xs rounded opacity-0 group-hover:opacity-100 transition-opacity pointer-events-none whitespace-nowrap z-50"
      >
        Eligibility
      </div>
    </button>
  </nav>
</div>
`;
